### PR TITLE
Update `CommonAnnotatedKey` fields scope to be accessed by derived classes

### PR DESCRIPTION
**Pull request by yongyan-gh**
## Description
- Update `CommonAnnotatedKey` private fields `bytes` and `base64Key` to properties with private setter, so their values can be accessed by derived classes.

### CODE REVIEW

File: docs/ReleaseHistory.md [hunk 1]
```diff
@@ -10,6 +10,9 @@
 - FPS => False positive reduction in static analysis.
 - FNS => Flase negative reduction in static analysis.
 
+# 1.4.22 - 05/21/2024
```
Comment (michaelcfanning) [inline docs/ReleaseHistory.md]: >05/21/2024 [](http://example.com/codeflow?start=11&length=10)

this should be UNRELEASED

Comment (michaelcfanning) [inline docs/ReleaseHistory.md]: >22 [](http://example.com/codeflow?start=6&length=2)

s/be 1.4.23

File: src/Microsoft.Security.Utilities.Core/CommonAnnotatedKey.cs [hunk 2]
```diff
@@ -59,8 +59,10 @@ public static bool TryCreate(string key, out CommonAnnotatedKey secret)
             return true;
         }
 
-        private byte[] bytes;
-        private string base64Key;
+        protected byte[] bytes;
```
Comment (michaelcfanning) [inline src/Microsoft.Security.Utilities.Core/CommonAnnotatedKey.cs]: >protected [](http://example.com/codeflow?start=8&length=9)

protected fields actually aren't legal according to the .NET guidelines!

Instead, create properties, can we just make the setter protected?

public string EncodedKey { get; protected set; }
public byte[] KeyBytes { get; protected set; }

Comment (yongyan-gh) [reply src/Microsoft.Security.Utilities.Core/CommonAnnotatedKey.cs]: got it

Comment (yongyan-gh) [reply src/Microsoft.Security.Utilities.Core/CommonAnnotatedKey.cs]: updated to properties with protected set

File: docs/ReleaseHistory.md (hunk 1 above)
Comment (yongyan-gh) [reply docs/ReleaseHistory.md]: done

Comment (yongyan-gh) [reply docs/ReleaseHistory.md]: updated to 1.4.23, correct?

Comment (michaelcfanning) [reply docs/ReleaseHistory.md]: Here's your entry. This is going into 1.4.23 but it didn't ship yesterday, it is unreleased. 

`1.4.23 - UNRELEASED`

File: src/Microsoft.Security.Utilities.Core/CommonAnnotatedKey.cs [hunk 3]
```diff
@@ -59,33 +59,36 @@ public static bool TryCreate(string key, out CommonAnnotatedKey secret)
             return true;
         }
 
-        private byte[] bytes;
-        private string base64Key;
+        public byte[] KeyBytes { get; protected set; }
```
Comment (michaelcfanning) [inline src/Microsoft.Security.Utilities.Core/CommonAnnotatedKey.cs]: >KeyBytes [](http://example.com/codeflow?start=22&length=8)

The setter for either of these properties should update the other form as well, arguably.

So if you set the bytes, rewrite the encoded key and vice versa. We should probably do that if we expect classes to extend this case class.

File: version.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "$schema": "https://raw.githubusercontent.com/dotnet/Nerdbank.GitVersioning/master/src/NerdBank.GitVersioning/version.schema.json",
-  "version": "1.4.22",
+  "version": "1.4.23",
```
Comment (michaelcfanning) [inline version.json]: >23 [](http://example.com/codeflow?start=18&length=2)

btw you don't actually need to publish this package to get this change, just update your submodule for now!

File: docs/ReleaseHistory.md (hunk 1 above)
Comment (yongyan-gh) [reply docs/ReleaseHistory.md]: done

File: src/Microsoft.Security.Utilities.Core/CommonAnnotatedKey.cs (hunk 3 above)
Comment (yongyan-gh) [reply src/Microsoft.Security.Utilities.Core/CommonAnnotatedKey.cs]: updated the setter logic to rewrite another property.

File: src/Microsoft.Security.Utilities.Core/CommonAnnotatedKey.cs [hunk 5]
```diff
@@ -59,33 +58,56 @@ public static bool TryCreate(string key, out CommonAnnotatedKey secret)
...
+        }
+
+        protected CommonAnnotatedKey() { }
 
         private CommonAnnotatedKey(byte[] bytes)
         {
-            this.bytes = bytes;
-            this.base64Key = Convert.ToBase64String(this.bytes);
+            this.keyBytes = bytes;
+            this.encodedKey = Convert.ToBase64String(this.keyBytes);
```
Comment (michaelcfanning) [inline src/Microsoft.Security.Utilities.Core/CommonAnnotatedKey.cs:90]: >ToBase64String [](http://example.com/codeflow?start=38&length=14)

where are your unit test for this change, friend? :)

**Comment by yongyan-gh** [comment]
close for now, if needed in future will revisit

**Pull request closed without merging**